### PR TITLE
HDDS-5431. compile both proto2 and proto3 versions of Client to OM interaction proto files.

### PR DESCRIPTION
**Pull request by mukul1987**
## What changes were proposed in this pull request?
For Ozone to work with both hadoop2 and pre hadoop 3.3 versions, it needs to compile both proto2 and proto3 versions of the protofiles

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-5431

## How was this patch tested?
Build succeeds and current tests pass

### CODE REVIEW

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/utils/VersionInfo.java [hunk 1]
```diff
@@ -85,8 +85,12 @@ public String getSrcChecksum() {
     return info.getProperty("srcChecksum", "Unknown");
   }
 
-  public String getHadoopProtocVersion() {
```
Comment (adoroszlai) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/utils/VersionInfo.java:88]: This class is marked `Public` and `Stable`, I guess we should keep this method for compatibility.

Comment (mukul1987) [reply hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/utils/VersionInfo.java:88]: done

File: hadoop-ozone/interface-client/pom.xml [hunk 2]
```diff
@@ -63,6 +63,59 @@ https://maven.apache.org/xsd/maven-4.0.0.xsd">
...
+                         value="org.apache.hadoop.thirdparty.protobuf"
+                         dir="target/generated-sources/protobuf/java/proto3/org/apache/hadoop/ozone/protocol/proto">
+                </replace>
+                <replace token="com.google.protobuf"
+                         value="org.apache.hadoop.thirdparty.protobuf"
+                         dir="target/generated-sources/protobuf/java/proto3/org/apache/hadoop/ozone/security/proto">
+                </replace>
+                <replace token="org.apache.hadoop.ozone.protocol.proto"
+                         value="org.apache.hadoop.ozone.protocol.proto3"
+                         dir="target/generated-sources/protobuf/java/proto3/org/apache/hadoop/ozone/protocol/proto">
```
Comment (adoroszlai) [inline hadoop-ozone/interface-client/pom.xml:100]: In addition to renaming the package, I think the files should be moved to `org/apache/hadoop/ozone/protocol/proto3`.

Comment (mukul1987) [reply hadoop-ozone/interface-client/pom.xml:100]: thanks, done

**Review by adoroszlai — APPROVED** [review]
Thanks @mukul1987 for updating the patch.  Latest version looks good in my IDE, too.

**Pull request merged**